shared/middleware/files_api.rb

```diff
@@ -404,6 +404,18 @@ def put_file(endpoint, encrypted_channel_id, filename, body)
       return bad_request if share_failure && share_failure[:type] != "address"
     end
 
+    # Don't allow project to be saved if it contains non-UTF-8 characters (causing error / project to not load when opened).
+    if 'sources' == endpoint
+      body_json = JSON.parse(body)
+      source = body_json["source"]
+      html = body_json["html"]
+
+      source_is_valid = source && has_valid_encoding?(source)
+      html_is_valid = html ? source.force_encoding("UTF-8").valid_encoding? : true # HTML only exists for AppLab projects
```

Comment:
Style Nit: Could we move the comment from the end of the line to it's own line between, like:
```
source_is_valid = ...
# comment
html_is_valid = ...
```
I think that helps contextualize the purpose of `html_is_valid` before we even get to it.

Reply:
agreed! will update